source/support.tex

```diff
@@ -3166,6 +3168,214 @@
+\item \tcode{strcmp(lhs.file_name(), rhs_p.file_name()) == 0}
+\item \tcode{strcmp(lhs.function_name(), rhs_p.function_name()) == 0}
+\item \tcode{lhs.line() == rhs_p.line()}
+\item \tcode{lhs.column() == rhs_p.column()}
+\end{itemize}
+
+\rSec2[support.srcloc.cons]{Creation}
+
+\begin{itemdecl}
+static consteval source_location current() noexcept;
+\end{itemdecl}
+\begin{itemdescr}
+\pnum
+\returns
+\begin{itemize}
+\item
+  When invoked by a function call
+  whose \grammarterm{postfix-expression} is
+  a (possibly parenthesized) \grammarterm{id-expression} naming \tcode{current},
+  returns a \tcode{source_location} with an implementation-defined value.
+  The value should be affected by \tcode{\#line}\iref{cpp.line}
+  in the same manner as for \tcode{__LINE__} and \tcode{__FILE__}.
+  The values of the exposition-only data members
+  of the returned \tcode{source_location} object
+  are indicated in \tref{support.srcloc.current}.
+
+\begin{libefftabvalue}
+  {Value of object returned by \tcode{current}}
+  {support.srcloc.current}
+\tcode{line_}    &
+  A presumed line number\iref{cpp.predefined}.
+  Line numbers are presumed to be 1-indexed;
+  however, an implementation is encouraged to use 0
+  when the line number is unknown. \\ \rowsep
+\tcode{column_}  &
+  An implementation-defined value denoting
+  some offset from the start of the line denoted by \tcode{line_}.
+  Column numbers are presumed to be 1-indexed;
+  however, an implementation is encouraged to use 0
+  when the column number is unknown. \\ \rowsep
+\tcode{file_name_} &
+  A presumed name of the current source file\iref{cpp.predefined} as an NTBS.
+  \\ \rowsep
+\tcode{function_name_} &
+  A name of the current function
+  such as in \tcode{__func__}\iref{dcl.fct.def.general} if any,
+  an empty string otherwise. \\
+\end{libefftabvalue}
+
+\item
+  Otherwise, that is, when invoked in some other way, returns a
+  \tcode{source_location} whose data members are initialized
+  with valid but unspecified values.
+\end{itemize}
+
+\pnum
+\remarks
+When a \grammarterm{brace-or-equal-initializer}
+is used to initialize a non-static data member,
+any calls to \tcode{current} should correspond to the location
+of the constructor or aggregate initialization that initializes the member.
+
+\pnum
+\begin{note}
+When used as a default argument\iref{dcl.fct.default},
+the value of the \tcode{source_location} will be
+the location of the call to \tcode{current} at the call site.
+\end{note}
+\end{itemdescr}
+
+\pnum
+\begin{example}
+\begin{codeblock}
+struct s {
+  source_location member = source_location::current();
+  int other_member;
+  s(source_location loc = source_location::current())
+    : member(loc)               // values of \tcode{member} will be from call-site
+  {}
+  s(int blather) :              // values of \tcode{member} should be hereabouts
+    other_member(blather)
+  {}
+  s(double)                     // values of \tcode{member} should be hereabouts
```

Comment:
Can we reword this?  values can't be "hereabouts".

Reply:
Updated.

Reply:
I believe "hereabouts" was intentional, because it's unspecified whether the location is the end of the constructor's _parameter-declaration-clause_, or the start/end of the _ctor-initializer_, or the opening brace of the _function-body_, or somewhere else thereabouts.

But I suppose "refer to this location" in the new wording is vague enough as well.